riker-macros/src/lib.rs

```diff
@@ -35,13 +35,13 @@ impl MsgTypes {
 
 impl Parse for MsgTypes {
     fn parse(input: ParseStream) -> Result<Self> {
-        let vars = Punctuated::<Ident, syn::token::Comma>::parse_terminated(input)?;
+        let vars = Punctuated::<PathSegment, syn::token::Comma>::parse_terminated(input)?;
 
         Ok(MsgTypes {
             types: vars
                 .into_iter()
                 .map(|t| MsgVariant {
-                    name: get_name(&t),
+                    name: get_name(&t.ident),
```

Comment:
The only replacement for this line I found during my tests is:
```rust
name: get_name(&t.path.segments.last().expect().ident),
```
I think there should be a way to use this without the unwrap, but in my tests I couldn't trigger the unwrap once. I think this is the case, because if no `PathSegment` is present, the parsing errors out before we actually get here.
So if we don't find any other solution, we could just leave the unwrap in here. But in this case it should be commented why it's here and why it can't fail.

Reply:
Another Option would be to change the `get_name` function to parse the whole path into a camel case message name, so using your example above the message name would then be `SomeModuleMessage`:
```
fn get_name(segments: &Punctuated<PathSegment, Colon2>) -> Ident {
    let vname = segments.iter().fold(String::new(), |mut acc, curr| {
        let ident = format!("{}", curr.ident);
        acc.push_str(&ident.split('_').map(|s| {
            let mut s = s.to_string();
            if let Some(c) = s.get_mut(0..1) {
                c.make_ascii_uppercase();
            }
            s
        }).collect::<String>());
        acc
    });
    syn::Ident::new(&vname, segments.span())
}
```
This would have the benefit that:
1. no unwrap is needed
2. there are no conflicting names if there are same message names in different paths

Reply:
I like the idea of changing the name function to work with the whole path. But for completeness we should pass the whole `TypePath` and handle `QSelf` as well, if present.

`QSelf` is used when a type is qualified like this: `<T as SomeTrait>::SomeType`. If I understand it correctly, that means that if we don't include the `QSelf`, the following two types would generate the same Name:
```rust
<A as SomeTrait>::SomeType
<B as SomeTrait>::SomeType
```
But I think this can also be handled in a separate issue/pull request.

Also, the `.span()` does not compile when I try it locally. Could you push the Changes for me to test?

Reply:
Okay I pushed the changes for the above.
I am looking right now into handling `QSelf` and implemented a first draft; will add a test of it and then push it for you to review

Reply:
Ok so formatting `Type` within `QSelf` is more of a pain then I expected since from what I saw I would have to match the different possible types and handle them, therefore if the priority for this is not that high, I'd leave it out for now.